config/routes.rb

```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+
+  resources :urls, only: %i[index create]
+  get '/:slug' => "urls#show"
```

Comment:
Make routes resourceful always. In this example, use slugs/:slug instead or /:slug